tests/by-util/test_cksum.rs

```diff
@@ -1277,3 +1277,27 @@ fn test_non_utf8_filename() {
         .stdout_is_bytes(b"SHA256 (funky\xffname) = e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\n")
         .no_stderr();
 }
+
+#[cfg(target_os = "linux")]
```

Comment:
why linux only ?

could you please to add a test to inject invalid utf8 ?

Reply:
> why linux only ?

This is somewhat related to [this](https://github.com/uutils/coreutils/pull/6563#issuecomment-2225260360).

TLDR: With the current MSRV being 1.70, on windows. there is no way to translate between `OsStr` and `&[u8]` without going through `str` (this holds for owned types as well).

> could you please to add a test to inject invalid utf8 ?

This is what the test should be doing. Using non UTF-8 characters that would make the old implementation fail because it would try to create `String`s from these characters.

Reply:
> there is no way to translate between OsStr and &[u8]

Yes there is: `os_str_as_bytes`, just like you also use in the crate itself.

> This is what the test should be doing.

No, the test does not check for correct handling of *files* with non-UTF-8 names.

Reply:
I'd like to be sure I get it correctly. In the added test, I should check that the `hashes` variable get written in a file with a non-UTF-8 name, is this correct ?

If yes, I might need to refactor slightly the test framework in order to be able to write to files with non-UTF-8 names.